### PR TITLE
Fix Cpu Requests priority Windows.

### PR DESCRIPTION
**Pull request by adelina-t**
For Windows, CPU Requests ( Shares, Count and Maximum ) are mutually exclusive, however
Kubernetes sends them all anyway in the pod spec.
When using dockershim this is not an issue, as Docker checks for this specific situation
here: https://github.com/moby/moby/blob/1bd184a4c291e4f60629e2cc279216f8f40495f3/daemon/daemon_windows.go#L87-L106

However, when using CRI-Containerd this pods fail to spawn with an error from hcsshim.

This PR intends to filter these values before they are sent to the CRI and not rely on the
runtime for it.

Related to: https://github.com/kubernetes/kubernetes/issues/84804

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/kubelet/kuberuntime/kuberuntime_container_windows.go [hunk 1]
```diff
@@ -82,6 +82,29 @@ func (m *kubeGenericRuntimeManager) generateWindowsContainerConfig(container *v1
...
+			if wc.Resources.CpuMaximum > 0 {
+				wc.Resources.CpuMaximum = 0
+				klog.Warningf("Mutually exclusive options: CPUCount priority > CPUMaximum priority on Windows Server Containers. CPUMaximum should be ignored")
+			}
+		} else if wc.Resources.CpuShares > 0 {
+			if wc.Resources.CpuMaximum > 0 {
+				wc.Resources.CpuMaximum = 0
+				klog.Warningf("Mutually exclusive options: CPUShares priority > CPUMaximum priority on Windows Server Containers. CPUMaximum should be ignored")
+			}
+
```
Comment (jterry75) [inline pkg/kubelet/kuberuntime/kuberuntime_container_windows.go]: NIT: extra line

Comment (adelina-t) [reply pkg/kubelet/kuberuntime/kuberuntime_container_windows.go]: Done

**Comment by jterry75** [comment]
LGTM (not a maintainer)

**Comment by PatrickLang** [comment]
cc @kubernetes/sig-node-pr-reviews  - this needs approve

File: pkg/kubelet/kuberuntime/kuberuntime_container_windows.go (hunk 1 above)
Comment (Random-Liu) [reply pkg/kubelet/kuberuntime/kuberuntime_container_windows.go]: It seems that this is not addressed.

**Comment by adelina-t** [comment]
/retest

Sent from my iPhone

On 7 Feb 2020, at 18:32, Kubernetes Prow Robot <notifications@github.com> wrote:

﻿

@adelina-t<https://github.com/adelina-t>: The following test failed, say /retest to rerun all failed tests:

Test name       Commit  Details Rerun command
pull-kubernetes-e2e-gce bc7d254<https://github.com/kubernetes/kubernetes/commit/bc7d254317cf64b64d29c05862c88d0da6c84474>       link<https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/85856/pull-kubernetes-e2e-gce/1225805504488411136/>   /test pull-kubernetes-e2e-gce

Full PR test history<https://prow.k8s.io/pr-history?org=kubernetes&repo=kubernetes&pr=85856>. Your PR dashboard<https://gubernator.k8s.io/pr/adelina-t>. Please help us cut down on flakes by linking to<https://git.k8s.io/community/contributors/devel/sig-testing/flaky-tests.md#filing-issues-for-flaky-tests> an open issue<https://github.com/kubernetes/kubernetes/issues?q=is:issue+is:open> when you hit one in your PR.

Instructions for interacting with me using PR comments are available here<https://git.k8s.io/community/contributors/guide/pull-requests.md>. If you have questions or suggestions related to my behavior, please file an issue against the kubernetes/test-infra<https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:> repository. I understand the commands that are listed here<https://go.k8s.io/bot-commands>.

**Pull request merged**